src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -284,12 +290,6 @@ private void CheckStakeKernelHash(PosRuleContext context, UnspentOutputs staking
 
             uint256 previousStakeModifier = this.GetPreviousStakeModifier(chainedHeader);
 
-            if (previousStakeModifier == null)
```

Comment:
OK so this just moved to `GetPreviousStakeModifier` method?

Reply:
the null check yes, even if looking at code we may still have a null if 
`this.LastCheckpoint.StakeModifierV2` is null (I don't expect a checkpoint having a null StakeModifierV2 tho, but maybe better add a check anyway in GetPreviousStakeModifier

Reply:
naa checkpoint must have a modifier specified (well otherwise it will throw anyway)